### ISSUE TITLE
Kubernetes API server as Rendezvous backend

### ISSUE THREAD

**Issue by sathyanarays** [pytorch/pytorch#121486]
### 🚀 The feature, motivation and pitch

The `torchrun` supports `etcd` and `c10d` as rendezvous backend. If torch run jobs need to be run in kubernetes clusters, we need to rely on `etcd` (either self hosted or hosted by the cloud provider). The Kubernetes API server is also based on `etcd`. If we can implement a `RendezvousHandler` that uses Kubernetes API server as rendezvous backend, we can run `torchrun` without any external dependencies on Kubernetes clusters.

Please provide your thoughts on this!

**Comment by d4l3k** [comment]
I don't have any estimation of how widely the etcd backend is used. In torchx we don't use etcd and instead use c10d with shared details. Many schedulers (slurm, volcano, etc) provide the rank 0 hostname on all ranks so can use that as the master_addr. 

With Kubernetes the DNS hostname is typically deterministic based off the job name so can be used instead of requiring etcd.

I'm happy to take contributions to improve etcd support or make it easier to use torchrun out of the box on Kubernetes though

**Comment by aojea** [comment]
> With Kubernetes the DNS hostname is typically deterministic based off the job name so can be used instead of requiring etcd.

yeah, let me give you some context on why we are chaming here, as a Kubernetes core dev, we consider PyTorch a Tier-1 workload, but we observe the integration is currently fragmented across many layers (Kubeflow, Ray, custom glue).

I want to help streamline the K8s/PyTorch relationship to remove these friction points rather than adding more layers, however, these ecosystem is vast and hard to navigate ... could you advise on the best starting point or contact to collaborate on this broader picture?
